gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -953,3 +953,120 @@ hook.Add("TTTEndRound", "TTTEndRound4TTT2TargetPlayer", function()
 		plys[i].targetPlayer = nil
 	end
 end)
+
+---
+-- Returns if a player is currently in a revival process started by @{Player:Revive}
+-- @return boolean The revival status
+-- @realm shared
+function plymeta:IsReviving()
+	return self:TTT2NETGetBool("player_is_reviving", false) and not self:Alive()
+end
+
+---
+-- Returns if the ongoing revival is blocking or not
+-- @return boolean The blocking status
+-- @realm shared
+function plymeta:IsBlockingRevival()
+	return self:TTT2NETGetBool("player_is_blocking_revival", false)
+end
+
+---
+-- Returns the time when the ongoing revival started.
+-- @return [default=@{CurTime()}]number The time when the revival started in seconds
+-- @realm shared
+function plymeta:GetRevivalStartTime()
+	return self:TTT2NETGetFloat("player_revival_start_time", CurTime())
+end
+
+---
+-- Returns the time for the ongoing revival.
```

Comment:
do you mean the full duration of the revival process? the remaining duration until the revive happens? an absolute point in time?